libs/core/src/lib/sources.spec.ts

```diff
@@ -0,0 +1,38 @@
+import { Subject } from 'rxjs';
+import { Effect } from './effect';
+import { setup } from './test-helper';
+
+it('sources data from subject', () => {
+  const value = { count$: new Subject() };
+  const effect: Effect<typeof value> = ({ sources }) => {
+    return sources.count$();
+  };
+  setup(
+    { value, effect },
+    ({ expectObservable, effect$, hot, storeEvent$ }) => {
+      hot('a').subscribe(value.count$);
+
+      expectObservable(effect$).toBe('a');
+      expectObservable(storeEvent$).toBe('(abcd)', {
+        a: { type: 'effect', name: 'root', event: 'spawn' },
+        b: {
+          type: 'link',
+          from: { type: 'subject', name: 'count$' },
+          to: { type: 'effect', name: 'root' },
+        },
+        c: {
+          type: 'value',
+          from: { type: 'subject', name: 'count$' },
+          to: { type: 'effect', name: 'root' },
+          value: 'a',
+        },
+        d: {
```

Comment:
I suppose there are more edge cases even though we're at 100% coverage, in a follow-up PR I can add variations of this test where the child does not allow the events to bubble to the parent effect, etc. to make sure we permute more of the edge cases.